nrich-search/src/test/java/net/croz/nrich/search/util/PathResolvingUtilTest.java

```diff
@@ -80,41 +85,52 @@ void shouldRemoveFirstPathElement() {
         assertThat(result).isEqualTo("second");
     }
 
+    @EnumSource(JoinType.class)
+    @NullSource
+    @ParameterizedTest
+    void shouldCalculateFullPath(JoinType joinType) {
+        // given
+        String firstPath = "restriction";
+        String secondPath = "attribute";
+        Path<?> expectedResult = mock(Path.class);
+        Root<?> first = createRootWithCollectionAttribute(firstPath, secondPath, expectedResult, joinType);
+
+        // when
+        Path<?> result = PathResolvingUtil.calculateFullPath(first, joinType, new String[] { firstPath, secondPath });
+
+        // then
+        assertThat(result).isEqualTo(expectedResult);
+    }
+
     @Test
-    void shouldCalculateFullRestrictionPath() {
+    void shouldReuseExistingJoinWhenCalculatingPath() {
```

Comment:
It's really hard to know what is going on here unless you deep dive into the helper method used in `given` block because mocking is done there. In my opinion, it was much easier to follow before introducing generating util.

Reply:
For me it seems that it can be clear by test name. I would prefer to keep it in generating util.